src/server/endpoint.cc

```diff
@@ -14,7 +14,7 @@ namespace Http {
 
 Endpoint::Options::Options()
     : threads_(1)
-    , flags_()
+    , flags_(Tcp::Options::InstallSignalHandler | Tcp::Options::ReuseAddr)
```

Comment:
This is a behaviour change.  I have mixed feelings about it.  I will likely merge this PR, but be prepared if others want this reverted.  However, IMHO, users should be explicitly setting flags.  I do know that at least one user of Pistache has their own SIGINT handler, and this change will break their code.  IDR who they were though.

What is your rationale for making this change?

Reply:
@dennisjenkins75 The rationale for making this change is the following:

1. I have seen several servers and most of time `SO_REUSEADDR` option is set when listen socket is configured., because you can stop server and run it immediately after that and it would be better have no problems with starting again by **default**, without this option.

2. I consider `Endpoint` as a high level (it means that we have made some desicion for users) building block in Pistache and it ok to handle by **default** CTRL + C to free resources. It looks natual. For instance, If I'm not mistaken in [Twisted](https://twistedmatrix.com/) reactor loop has default `SIGINT` handler. On the other hand, if you know users that use their own `SIGINT` handler, it would be better to back out my change.

Ok, I will revert the part of commit that is related to `Endpoint` soon.